### PR TITLE
Alex Gallant

### PR DESCRIPTION
**Pull request by agallant121**
Could not figure out join on 3

### CODE REVIEW

File: app/models/dish.rb [hunk 1]
```diff
@@ -1,4 +1,15 @@
 class Dish <ApplicationRecord
   validates_presence_of :name, :description
+
   belongs_to :chef
+  has_many :ingredient_dishes
+  has_many :ingredients, through: :ingredient_dishes
+
+  def total_calories
+    ingredients.sum(:calorie_count)
```
Comment (mikedao) [inline app/models/dish.rb]: Good

File: spec/features/dish/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
```
Comment (mikedao) [inline spec/features/dish/show_spec.rb]: As we discussed in the review, would like to have seen another chef slash dish

File: spec/models/dish_spec.rb [hunk 3]
```diff
@@ -7,5 +7,20 @@
   end
   describe "relationships" do
     it {should belong_to :chef}
+    it {should have_many :ingredient_dishes}
+    it {should have_many(:ingredients).through(:ingredient_dishes)}
   end
+
+  it "instance method" do
```
Comment (mikedao) [inline spec/models/dish_spec.rb]: instance method should be in the describe block and the it block should be checking for the actual method

**Comment by agallant121** [comment]
Yeah. I was working on that joins for the last 40 min and kept moving it
around to see where it would work. I knew it had to be in the chef but did
not know how to get it to the ingredients.

**Pull request closed without merging**